hugolib/site.go

```diff
@@ -343,8 +343,19 @@ func (s *SiteInfo) SourceRelativeLinkFile(ref string, currentPage *Page) (string
 			return "", fmt.Errorf("No file found for \"%s\" on page \"%s\".\n", ref, currentPage.Source.Path())
 		}
 
-		link = target.Path()
-		return "/" + filepath.ToSlash(link), nil
+		link = filepath.ToSlash(target.Path())
+		if viper.GetBool("CanonifyURLs") {
+			return "/" + link, nil
```

Comment:
@bep / @anthonyfok from what I have gleaned from the docs, `CanonifyURLs` should mean that the baseUrl _is_ added?

https://gohugo.io/extras/urls#canonicalization

interestingly, when I fix my SourceRelativeLink() to _not_  unintentionally add the `baseUrl` path to pages, `CanonifyURLs=true` starts fixing up abs urls by adding them. (yeah, convoluted)